python/ray/tune/trial_runner.py

```diff
@@ -744,6 +565,19 @@ def _process_trial(self, trial):
                 self._scheduler_alg.on_trial_complete(self, trial, flat_result)
                 self._search_alg.on_trial_complete(
                     trial.trial_id, result=flat_result)
+
+                result.update(done=True)
```

Comment:
can we move the `result.update` into L565 or somewhere higher? this is just so that search_alg and scheduler_alg will see the same set of values as the callback.

Reply:
I deliberately put this under the callbacks, as they didn't see the `done` tag before (it was added in `Trial.update_last_result`). But I agree, we should keep this the same. Moved.